search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
```

Comment:
A silly thing, but something I noticed. All other endpoints are pluralized.  Even the Browse endpoint is named `browses`.  So, that makes me wonder if we should name this `searches` (although I admit that sounds awkward to me as well).  Thoughts?